server/pages/index.js

```diff
@@ -319,6 +319,13 @@ module.exports = function() {
 
 	app.get( '/theme', ( req, res ) => res.redirect( '/design' ) );
 
+	app.get( '/plans', function( req, res, next ) {
```

Comment:
Nice catch. However, it will be called only on `/plans` page, not every page as you said :D. But we still need the `! req.cookies.wordpress_logged_in` on each `/plans` page call.